subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/ErrorHandlingModuleComponentRepository.java

```diff
@@ -94,15 +100,30 @@ public ModuleComponentRepositoryAccess getRemoteAccess() {
     }
 
     private static final class ErrorHandlingModuleComponentRepositoryAccess implements ModuleComponentRepositoryAccess {
+        private static final Logger LOGGER = Logging.getLogger(ErrorHandlingModuleComponentRepositoryAccess.class);
+        private final static String MAX_RETRIES_BEFORE_BLACKLISTING = "org.gradle.internal.repository.max.retries";
+        private final static String INITIAL_BACKOFF_MS = "org.gradle.internal.repository.initial.backoff";
+
         private final static String BLACKLISTED_REPOSITORY_ERROR_MESSAGE = "Skipped due to earlier error";
+
         private final ModuleComponentRepositoryAccess delegate;
         private final String repositoryId;
         private final RepositoryBlacklister repositoryBlacklister;
+        private final int maxRetriesCount;
+        private final int initialBackOff;
 
         private ErrorHandlingModuleComponentRepositoryAccess(ModuleComponentRepositoryAccess delegate, String repositoryId, RepositoryBlacklister repositoryBlacklister) {
+            this(delegate, repositoryId, repositoryBlacklister, Integer.getInteger(MAX_RETRIES_BEFORE_BLACKLISTING, 3), Integer.getInteger(INITIAL_BACKOFF_MS, 125));
+        }
+
+        private ErrorHandlingModuleComponentRepositoryAccess(ModuleComponentRepositoryAccess delegate, String repositoryId, RepositoryBlacklister repositoryBlacklister, int maxRetriesCount, int initialBackoff) {
+            assert maxRetriesCount > 0 : "Max retries must be > 0";
+            assert initialBackoff >= 0 : "Initial backoff must be >= 0";
```

Comment:
Does it really make sense to allow `backoff` values of 0?

Reply:
Yes, for tests.